### ISSUE TITLE
[Bug]: java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String issue with Boolean field

### ISSUE THREAD

**Issue by lifesucks6000** [apache/beam#22486]
### What happened?

**Beam version: 2.30**
**Flink version: 1.12.5**

When I tried to run my beam pipeline with **FlinkRunner**, I got this error -
`java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String`

Although when I ran it with beam **DirectRunner**, it ran fine with no error.

The field type on which this issue is occurring is of **Boolean** type. While reading from SQLServer 2008 DB, a Boolean field is treated as **Logical** type field. Here we are building the schema with the help of this method - 

`Schema.Field booleanField  = Schema.Field.nullable(name,Schema.FieldType.BOOLEAN);`

I managed to find a work around for this error with a slight change in the way I was building my schema : 

`Schema.Field booleanField = Schema.Field.nullable(name,Schema.FieldType.of(Schema.TypeName.BOOLEAN));`

But shouldn't both of these code changes would have worked in the same way? The former has worked for me in previous versions.
I think there is some compatibility issue with Apache Beam and FlinkRunner on Boolean data type

### Issue Priority

Priority: 2

### Issue Component

Component: runner-flink

**Comment by je-ik** [comment]
Can you attach some simple code to reproduce this?

**Comment by kennknowles** [comment]
Yes, or even a full stack trace so we know where this is occurring. But code to reproduce would be best if we are to address this.
